src/test/java/mmosii/bookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,123 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import mmosii.bookstore.dto.shoppingcart.CartItemDto;
+import mmosii.bookstore.dto.shoppingcart.CartItemRequestDto;
+import mmosii.bookstore.dto.shoppingcart.CartItemUpdateDto;
+import mmosii.bookstore.dto.shoppingcart.ShoppingCartDto;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Sql(scripts = {"classpath:database/add-shopping-cart-and-user.sql",
+        "classpath:database/add-books-and-categories-to-books-table.sql"})
+@Sql(scripts = {"classpath:database/delete-shoppingcarts-and-user.sql",
+        "classpath:database/delete-books-and-categories-from-books-table.sql"},
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+public class ShoppingCartControllerTest {
+    protected static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @DisplayName("Get shopping cart for valid user")
+    @WithMockUser(username = "test@gmail.com", authorities = {"USER", "ADMIN"})
+    void getShoppingCart_validUser_returnsDto() throws Exception {
+        ShoppingCartDto expected = new ShoppingCartDto(3L, 5L, null);
+
+        MvcResult result = mockMvc.perform(get("/api/cart")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        ShoppingCartDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), ShoppingCartDto.class);
+        assertThat(actual).isNotNull();
+        assertThat(actual.id()).isEqualTo(expected.id());
+        assertThat(actual.userId()).isEqualTo(expected.userId());
```

Comment:
You can inline it using one assertThat 
```suggestion
        assertThat(actual).isNotNull()
        .hasFieldOrPropertyWithValue("id", expected.id())
       .hasFieldOrPropertyWithValue("userId", expected.userId());
```
Something like this